packages/cloud/src/libraries/tenants.ts

```diff
@@ -67,6 +67,45 @@ export class TenantsLibrary {
     });
   }
 
+  async deleteTenantById(tenantId: string) {
+    const {
+      deleteTenantById,
+      deleteDatabaseRoleForTenant,
+      deleteResourcesByTenantId,
+      deleteRoleByTenantId,
+      getTenantById,
+      deleteM2MApplicationByTenantId,
+      removeUrisFromAdminConsoleRedirectUris,
+    } = this.queries.tenants;
+    const { cloudUrlSet } = EnvSet.global;
+
+    // `dbUser` is defined as nullable but we always specified this value when creating a new tenant.
+    const { dbUser } = await getTenantById(tenantId);
+    if (dbUser) {
+      await deleteDatabaseRoleForTenant(dbUser); // DB role for building connection for the current tenant.
+    }
+
+    await deleteTenantById(tenantId);
+
+    /**
+     * All applications, resources, scopes and roles attached to the current tenant
+     * will be deleted per DB design.
+     * Need to manually delete following applications, roles, resources since they
+     * are created for admin tenant which will not be deleted automatically.
+     */
+    // Delete management API for the current tenant.
+    // `scopes` will be automatically deleted if its related resources have been removed.
+    await deleteResourcesByTenantId(tenantId);
+    // Delete admin tenant admin role for the current tenant.
+    await deleteRoleByTenantId(tenantId);
```

Comment:
I don't quite get this one

Reply:
This role enables the current tenant to access its own management API.
Other roles assigned to the current tenant will be removed automatically upon the removal of the current tenant. But this role is with `admin` tenant, so we should manually delete this.